### PR TITLE
AAB(Android App Bundle) Support

### LINKED ISSUE #594: Google Play does not accept .apk anymore.
Now, we are now forced to upload an .aab (Android App Bundle) to Google Play.

### PR DESCRIPTION
**Pull request by rupesh-kumar-lpu**
Fixes #594 

As AAB is the only way to distribute apps on Google Play Console. This PR adds a feature to generate signed AAB and Unsigned AAB. For Testing purposes, a universal APK was created with the help of bundletool.jar(included in SDK) from generated AAB file and installed on the emulator. No issue has been faced by doing so.

**Generates AAB files as-**
`Documents/Processing/sketch_210923a/buildBundle/sketch_210923a_release_signed.aab`
`Documents/Processing/sketch_210923a/buildBundle/sketch_210923a_release_unsigned.aab`

**Regenerates Bundles similar to export-package utility as -**
`Documents/Processing/sketch_210923a/buildBundle.210924.0527/sketch_210923a_release_signed.aab`
`Documents/Processing/sketch_210923a/buildBundle.210924.0527/sketch_210923a_release_unsigned.aab`

**Method of initiating the task**
     `File -> Export Signed Bundle`

**Keyboard Short Cut**
`Ctrl + B`

Capture

### CODE REVIEW

File: mode/src/processing/mode/android/Manifest.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
   static final String MANIFEST_ERROR_MESSAGE =
     "Errors occurred while reading or writing " + MANIFEST_XML + ",\n" +
     "which means lots of things are likely to stop working properly.\n" +
-    "To prevent losing any data, it's recommended that you use “Save As”\n" +
+    "To prevent losing any data, it's recommended that you use Save As\n" +
```
Comment (ranaaditya) [inline mode/src/processing/mode/android/Manifest.java:51]: Save As was highlighted please revert the change here.

**Review by ranaaditya — CHANGES_REQUESTED** [review]

**Comment by ranaaditya** [comment]
Thanks for this amazing contribution @rupesh-kumar-lpu :partying_face: 

Do you have any kind of high level design doc for this feature or a rough idea how you have done this with the existing processing-android codebase ?

Thanks.

File: mode/src/processing/mode/android/Manifest.java (hunk 1 above)
Comment (rupesh-kumar-lpu) [reply mode/src/processing/mode/android/Manifest.java]: Yes, I don't know why but an unknown error was faced on this line due to these two characters used to highlight. After removing these characters Gradle's build was successful. That can be somewhat related to IDE saving the file in unknown encoding. I will confirm if the same error encounters again.

I am reverting the changes for now as you suggested. Thanks for pointing it out!

**Comment by rupesh-kumar-lpu** [comment]
For implementing this I tried to reuse the existing functions. The implementation is pretty much similar to the "Export Signed Package". The origin of implementation starts from the 'JMenuItem' click handler in the AndroidEditor.java file. All the functions used for "Export Signed Package" were reused with slight modification(providing certain function parameters) to distinguish the use of AAB or APK tasks. For example, KeystoreManager's constructor was modified to pass the boolean type (forBundle) to distinguish about the exporting task.

Thanks for reviewing the changes @ranaaditya

**Review by ranaaditya — APPROVED** [review]

**Comment by codeanticode** [comment]
@rupesh-kumar-lpu thanks for the PR, will review and merge as soon as I can!

**Comment by codeanticode** [comment]
@rupesh-kumar-lpu Finally merged, thank you!

**Comment by rupesh-kumar-lpu** [comment]
Thank you for taking out time to read too many lines of code 😊 @codeanticode @ranaaditya
